messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/ActiveMQServerService.java

```diff
@@ -304,16 +302,11 @@ public synchronized void start(final StartContext context) throws StartException
                 for(final Map.Entry<String, DiscoveryGroupConfiguration> entry : discoveryGroups.entrySet()) {
                     final String name = entry.getKey();
                     final String key = "discovery" + name;
-                    DiscoveryGroupConfiguration config = null;
+                    final DiscoveryGroupConfiguration config;
                     if (jgroupFactories.containsKey(key)) {
                         ChannelFactory channelFactory = jgroupFactories.get(key);
                         String channelName = jgroupsChannels.get(key);
-                        JChannel channel = channels.get(channelName);
-                        if (channel == null) {
-                            channel = (JChannel) channelFactory.createChannel(channelName);
-                            channels.put(channelName, channel);
-                        }
-                        config = DiscoveryGroupAdd.createDiscoveryGroupConfiguration(name, entry.getValue(), channel, channelName);
+                        config = DiscoveryGroupAdd.createDiscoveryGroupConfiguration(name, entry.getValue(), channelFactory, channelName, channels);
```

Comment:
You are right about the intent of the previous logic. However I was not matching Artemis expectations (which creates channels as it needs it without sharing them between its broadcast and discovery groups).

I'm not sure there is a correct use case for sharing the same channel name between discovery groups. The discovery groups would only differ by their `refresh-timeout` and `initial-wait-timeout`attributes. I don't see a compelling use case for this but it's technically acceptable...